### PR TITLE
docs: fix release date in changelogs

### PR DESCRIPTION
**Pull request by sarahgm**
# Description

Fixed the dates, with using github api, instead of simple-git. Because docs branch cannot read it

# Reviewers:

@marigold-ui/developer

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -25,6 +25,7 @@ out
 .component-props
 
 ## env
+docs/.env
```
Comment (sebald) [inline .gitignore:28]: Why is this necessary. the local env should be enough, no?

File: docs/scripts/build-changelog.mjs [hunk 2]
```diff
@@ -1,8 +1,9 @@
+import { Octokit } from '@octokit/core';
 import path from 'path';
-import { simpleGit } from 'simple-git';
 import { fs, globby } from 'zx';
 
-const git = simpleGit();
+require('dotenv').config();
```
Comment (sebald) [inline docs/scripts/build-changelog.mjs:5]: Maybe change to also use `.env.local` so we don't have multiple env files?

**Review by sebald — CHANGES_REQUESTED** [review]

File: docs/scripts/build-changelog.mjs (hunk 2 above)
Comment (sebald) [reply docs/scripts/build-changelog.mjs:5]: (Or does this not work in vercel then?)

File: .gitignore (hunk 1 above)
Comment (sarahgm) [reply .gitignore:28]: no didn't work for me

File: docs/scripts/build-changelog.mjs (hunk 2 above)
Comment (sarahgm) [reply docs/scripts/build-changelog.mjs:5]: For me it doesn't work with the local env file

Comment (sebald) [reply docs/scripts/build-changelog.mjs:5]: But why? You can specify a different .env file for the lib.

Comment (sarahgm) [reply docs/scripts/build-changelog.mjs:5]: my token is than undefined

File: docs/scripts/build-changelog.mjs [hunk 3]
```diff
@@ -12,27 +13,68 @@ let changelogPath = await globby([
 
 console.log('📑 Generating changelogs...');
 
-const getReleaseInformation = async file => {
-  const log = await git.log({ file: path.resolve(file) });
+const getReleaseData = async () => {
```
Comment (aromko) [inline docs/scripts/build-changelog.mjs:16]: I asked Copilot to simplify this code snippet from l.16-77 and thats the result

```
const getReleaseData = async () => {
  const octokit = new Octokit({ auth: GITHUB_TOKEN });
  const response = await octokit.request('GET /repos/marigold-ui/marigold/releases', {
    headers: { 'X-GitHub-Api-Version': '2022-11-28' },
  });

  return response.data.reduce((releaseDates, release) => {
    const [, version] = release.name.match(/@([^@]+)$/) || [];
    if (!version) return releaseDates;

    if (!releaseDates.has(version)) {
      const releaseDate = new Date(release.published_at);
      const daysDifference = (new Date() - releaseDate) / (24 * 60 * 60 * 1000);
      releaseDates.set(version, {
        releaseDate,
        badge: daysDifference < 30 ? 'new' : undefined,
      });
    }
    return releaseDates;
  }, new Map());
};

const getReleaseInformation = (sourceText, releaseDates) => {
  const versionRegex = /## (\d{1,2}\.\d{1,2}\.\d{1,2})/gm;
  const versions = sourceText.match(versionRegex)?.map(v => v.replace('## ', '')) || [];

  if (!versions.length) {
    console.log('No versions found in the source text.');
    return [];
  }

  return versions.map(version => ({
    version,
    releaseDate: releaseDates.get(version)?.releaseDate || null,
    badge: releaseDates.get(version)?.badge || undefined,
  }));
```

Changes made:

- Combined the Map initialization and population in one step within reduce.
- Removed unnecessary intermediate variables.
- Simplified the calculation of daysDifference.
- Used optional chaining for safer access in getReleaseInformation.

**Comment by sebald** [comment]
Can we remove the changelog files from git in this PR too?

File: docs/scripts/build-changelog.mjs (hunk 3 above)
Comment (sebald) [reply docs/scripts/build-changelog.mjs:16]: can you ask him to not use `.reduce`? :D I always find this makes the code harder to comprehend

Comment (sebald) [reply docs/scripts/build-changelog.mjs:16]: I gave this chatGPT 4o:


```js
const getReleaseData = async () => {
  const octokit = new Octokit({ auth: GITHUB_TOKEN });
  const response = await octokit.request('GET /repos/marigold-ui/marigold/releases', {
    headers: {
      'X-GitHub-Api-Version': '2022-11-28',
    },
  });

  const releases = response.data;
  const releaseDates = new Map();

  releases.forEach((release) => {
    const versionMatch = release.name.match(/@([^@]+)$/);
    const version = versionMatch ? versionMatch[1] : null;
    if (!version) return;

    if (!releaseDates.has(version)) {
      const releaseDate = new Date(release.published_at);
      const daysSinceRelease = calculateDaysSince(releaseDate);
      const badge = daysSinceRelease < 30 ? 'new' : undefined;

      releaseDates.set(version, {
        releaseDate,
        badge,
      });
    }
  });

  return releaseDates;
};

const calculateDaysSince = (date) => {
  const today = new Date();
  const timeDifference = today - date;
  const millisecondsInADay = 24 * 60 * 60 * 1000;
  return Math.round(timeDifference / millisecondsInADay);
};
```

### Changes Made:
1. **Extracted Regular Expression Match**: Stored the regex match result in `versionMatch` to simplify `version` extraction and improve readability.
2. **Renamed Variables for Clarity**:
   - Changed `aDayInMs` to `millisecondsInADay` for better understanding.
3. **Extracted Day Calculation Logic**: Moved the calculation of the number of days since the release into a separate function, `calculateDaysSince`, for clarity and reuse.
4. **Simplified Badge Logic**: Directly assigned `badge` during the calculation.

File: docs/scripts/build-changelog.mjs (hunk 2 above)
Comment (sarahgm) [reply docs/scripts/build-changelog.mjs:5]: how can I test it in vercel? I added a env variable in vercel for it. And the env file from me is created in docs/.env

Comment (sarahgm) [reply docs/scripts/build-changelog.mjs:5]: https://marigold-docs-3meutb371-marigold.vercel.app/releases/packages/components/release?theme=core

File: docs/scripts/build-changelog.mjs (hunk 3 above)
Comment (sarahgm) [reply docs/scripts/build-changelog.mjs:16]: This is exactly the same except you moved. the calculatoin of date difference out of it right?

Comment (sarahgm) [reply docs/scripts/build-changelog.mjs:23]: I changed it can you have a look again?

File: docs/scripts/build-changelog.mjs (hunk 2 above)
Comment (sebald) [reply docs/scripts/build-changelog.mjs:5]: https://www.npmjs.com/package/dotenv#path

```
require('dotenv').config({ path: ['.env.local', '.env'] })
```

Comment (sarahgm) [reply docs/scripts/build-changelog.mjs:5]: I just have one env file.....

Comment (sarahgm) [reply docs/scripts/build-changelog.mjs:5]: By default, config will look for a file called .env in the current working directory.

**Review by sebald — APPROVED** [review]
